tests/TwoFactor/ConfirmFormTest.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace Wearesho\Yii2\Authentication\Tests\TwoFactor;
+
+use PHPUnit\Framework\TestCase;
+use Wearesho\Yii\Http;
+use Wearesho\Yii2\Authentication;
+use Wearesho\Yii2\Authentication\Tests\Mock;
+use Wearesho\Yii2\Authorization;
+use Wearesho\Yii2\Token;
+use yii\web;
+
+/**
+ * Class ConfirmFormTest
+ * @package Wearesho\Yii2\Authentication\Tests\TwoFactor
+ */
+class ConfirmFormTest extends TestCase
+{
+    /**
+     * @expectedException \yii\base\InvalidConfigException
+     * @expectedExceptionMessage Invalid data type: stdClass. Wearesho\Yii2\Token\Repository is expected.
+     */
+    public function testInvalidTokenRepositoryDependency(): void
+    {
+        new Authentication\TwoFactor\ConfirmForm(
+            $this->createMock(Http\Request::class),
+            $this->createMock(Http\Response::class),
+            [
+                'tokenRepository' => new \stdClass,
+            ]
+        );
+    }
+
+    /**
+     * @expectedException \yii\base\InvalidConfigException
+     * @expectedExceptionMessage Invalid data type: stdClass. Wearesho\Yii2\Authorization\Repository is expected.
+     */
+    public function testInvalidAuthorizationRepositoryDependency(): void
+    {
+        new Authentication\TwoFactor\ConfirmForm(
+            $this->createMock(Http\Request::class),
+            $this->createMock(Http\Response::class),
+            [
+                'tokenRepository' => $this->createMock(Token\Repository::class),
+                'authorizationRepository' => new \stdClass,
+            ]
+        );
+    }
+
+    public function testMissingHash(): void
+    {
+        $form = new Authentication\TwoFactor\ConfirmForm(
+            $this->createMock(Http\Request::class),
+            $this->createMock(Http\Response::class),
+            [
+                'tokenRepository' => $this->createMock(Token\Repository::class),
+                'authorizationRepository' => $this->createMock(Authorization\Repository::class),
+            ]
+        );
+        $this->assertFalse($form->validate(['hash']));
+    }
+
+    public function testMissingValue(): void
+    {
+        $form = new Authentication\TwoFactor\ConfirmForm(
+            $this->createMock(Http\Request::class),
```

Comment:
What is the point of these mocks, if you do not specify any method

Reply:
The purpose is to avoid creating real object and configuring it